xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/EmbeddedSolr.java

```diff
@@ -463,9 +470,26 @@ private void createSearchCore() throws IOException
     private void createCacheCore(Path corePath, String solrCoreName) throws IOException
     {
         // Indicate the path of the data
-        Path dataPath = getCacheCoreDataDir(corePath, solrCoreName);
-        FileUtils.write(corePath.resolve("core.properties").toFile(), "dataDir=" + dataPath, StandardCharsets.UTF_8,
-            true);
+        Path dataDir = getCacheCoreDataDir(corePath, solrCoreName);
+        File corePropertiesFile = corePath.resolve(CORE_PROPERTIES_FILENAME).toFile();
+        Properties coreProperties = new Properties();
+        // we used to append this property to this file.
+        // I'm not sure any other properties are ever in here, but jic let's be passive
```

Comment:
It's very unlikelly since #createCacheCore is supposed to be used only to create a new cache.

Reply:
Sure, but it's possible someone familiar with Solr decided to put more options in there I think? This would get called if I delete the "<perm>/cache" directory then? I do like that we clobber this property in this case because it means the cache directory location will get fixed when people upgrade. 

I could check for a bugged property. If the property doesn't contain File.separator  I could assume it is bugged and reset it to the default, but otherwise leave it alone in case someone decided to put their cache somewhere else. Just an idea.

Reply:
If we wanted to get really fancy we could move the cache from the wrong location to the right location.

Reply:
I'm noticing there is precedent in the class for doing clean up of old locations and files, so I think I should do that. Updates forthcoming.

Reply:
> Sure, but it's possible someone familiar with Solr decided to put more options in there I think?

It's not really a use case which is supported right now, the search core configuration is wiped everytime something is changed to the schema that require a reindex.
So currently, when createSearchCore() is called, the Solr core configuration does not exist (either it did not exist in the first place or it was deleted because it's not updated anymore).